components/brave_shields/browser/ad_block_regional_service_manager.cc

```diff
@@ -261,13 +261,17 @@ absl::optional<base::Value> AdBlockRegionalServiceManager::UrlCosmeticResources(
   }
   absl::optional<base::Value> first_value =
       it->second->UrlCosmeticResources(url);
+  DCHECK(first_value->is_dict());
 
   for (; it != regional_services_.end(); it++) {
     absl::optional<base::Value> next_value =
         it->second->UrlCosmeticResources(url);
+    DCHECK(next_value->is_dict());
+
     if (first_value) {
       if (next_value) {
-        MergeResourcesInto(std::move(*next_value), &*first_value, false);
+        MergeResourcesInto(std::move(next_value->GetDict()),
+                           first_value->GetIfDict(), false);
```

Comment:
What's the reason to use `GetIfDict()` here? We already `DCHECK` that it `is_dict`.

Noticed the same pattern in a few other places as well.

Reply:
I have used `GetIfDict()` merely because `MergeResourcesInto` expects a pointer as a second argument.